### PR TITLE
docs: improve partitioning docs clarity

### PR DESCRIPTION
**Pull request by ajeddeloh**
with feedback from https://github.com/coreos/ignition/pull/544#pullrequestreview-125013424

### CODE REVIEW

**Comment by dgonyeo** [comment]
retest this please

File: doc/operator-notes.md [hunk 1]
```diff
@@ -81,20 +81,20 @@ The following table shows the possible combinations of whether or not a partitio
...
 
 ### Partition start 0
-Specifying `start` as 0 will use the starting sector of the largest available block. It will *not* use the first available block large enough.
+Specifying `start` as 0 will use the starting sector of the largest available block. This is not necessarily the first available block large enough.
 
 ### Unspecified partition start
-If `start` is not specified and a partition with the same number exists, it will use the value of the existing partition, unless wipePartitionEntry is set.
-If `start` is not specified and there is no existing partition, or wipePartitionEntry is set, `start` act as if it were set to 0 and use the starting sector of the largest available block.
+If `start` is not specified and a partition with the same number exists, Ignition will use the start of the existing partition, unless wipePartitionEntry is set.
+If `start` is not specified and there is no existing partition, or wipePartitionEntry is set, Ignition with use the starting sector of the largest block, as if `start` were set to 0.
```
Comment (dgonyeo) [inline doc/operator-notes.md]: s/with/will/

File: doc/configuration-v2_3-experimental.md [hunk 2]
```diff
@@ -28,16 +28,15 @@ The Ignition configuration is a JSON document conforming to the following specif
...
       * **_size_** (integer): the size of the partition (in device logical sectors, 512 or 4096 bytes). If zero, the partition will be made as large as possible.
       * **_start_** (integer): the start of the partition (in device logical sectors). If zero, the partition will be positioned at the start of the largest block available.
       * **_typeGuid_** (string): the GPT [partition type GUID][part-types]. If omitted, the default will be 0FC63DAF-8483-4772-8E79-3D69D8477DE4 (Linux filesystem data).
       * **_guid_** (string): the GPT unique partition GUID.
-      * **_wipePartitionEntry_** (boolean) whether or not Ignition is allowed to wipe an existing partition entry for this partition number.
-      * **_shouldExist_** (boolean) whether or not a partition `number` should exist. If ommited, it defaults to true. If false, `number` must be specified and non-zero and `label`, `start`, `size`, `guid`, and `typeGuid` must all be omitted.
+      * **_wipePartitionEntry_** (boolean) if true, Ignition will clobber an existing partition if it does not match the config. If false, Ignition will fail instead.
```
Comment (lucab) [inline doc/configuration-v2_3-experimental.md]: Perhaps this should specify what is the default behavior? (If there isn't anything nearby already specifying it)

**Comment by ajeddeloh** [comment]
@lucab does that clear things up?

**Comment by lucab** [comment]
@ajeddeloh yep, thanks, LGTM.

**Pull request merged**